helper/acctest/random.go

```diff
@@ -134,47 +132,33 @@ func RandIpAddress(s string) (string, error) {
 		return prefix.Addr().String(), nil
 	}
 
-	prefixSizeExponent := uint(prefix.Addr().BitLen() - prefix.Bits())
-
-	if prefix.Addr().Is4() && prefixSizeExponent > 31 {
-		return "", fmt.Errorf("CIDR range is too large: %d", prefixSizeExponent)
+	// base address as byte slice
+	prefixBytes, err := prefix.Addr().MarshalBinary()
+	if err != nil {
+		return "", err
 	}
 
-	// Prevent panics with rand.Int63n().
-	if prefix.Addr().Is6() && prefixSizeExponent > 63 {
-		return "", fmt.Errorf("CIDR range is too large: %d", prefixSizeExponent)
+	// inverse mask (ones in the host bits) as byte slice
+	inverseMaskBytes, err := inverseMask(prefix.Bits(), len(prefixBytes))
+	if err != nil {
+		return "", err
 	}
 
-	// Calculate max random integer based on the prefix.
-	// Bit shift 1<<size and subtract 1 to not overflow.
-	// e.g. 1<<8 - 1 = 256 - 1 = 255 for 192.168.0.0/24
-	randIntMax := big.NewInt(1)
-	randIntMax.Lsh(randIntMax, prefixSizeExponent)
-	randIntMax.Sub(randIntMax, big.NewInt(1))
-
-	// Prevent panics with rand.Int63n().
-	if randIntMax.Cmp(big.NewInt(0)) <= 0 {
-		return prefix.Addr().String(), nil
+	// the result starts life as 4 or 16 bytes of random data
+	resultBytes := make([]byte, len(inverseMaskBytes))
+	_, err = crand.Read(resultBytes)
+	if err != nil {
+		return "", err
 	}
 
-	randInt := rand.Int63n(randIntMax.Int64())
-
-	if randInt == 0 {
-		return prefix.Addr().String(), nil
+	// use the prefix and inverse mask to restore the network bits
+	for i := range inverseMaskBytes {
+		resultBytes[i] = (resultBytes[i] & inverseMaskBytes[i]) + prefixBytes[i]
 	}
 
-	// Calculate random address by taking prefix address and adding the random
-	// integer.
-	randAddrInt := new(big.Int).SetBytes(prefix.Addr().AsSlice())
-	randAddrInt.Add(randAddrInt, big.NewInt(randInt))
-
-	randAddr, ok := netip.AddrFromSlice(randAddrInt.Bytes())
+	result, _ := netip.AddrFromSlice(resultBytes)
```

Comment:
nit: I know it's not possible at this point, given the logic above, but I feel like we should keep the original logic of checking the `ok` return from `netip.AddrFromSlice` and returning an error with the byte data:

```go
result, ok := netip.AddrFromSlice(resultBytes)
if !ok {
    return "", fmt.Errorf("unexpected error creating random address from bytes: %#v", resultBytes)
}
```

Reply:
Hey @austinvalle,

I think I took care of your punch list.

Additionally, see the inline comment about the addition of `Masked()` (a change you didn't request, but seems like a better way to handle slightly-broken input).

Thanks!